workspace/workspace.go

```diff
@@ -113,7 +113,10 @@ func (ws Workspace) SolutionDir(s string) (string, error) {
 		if _, err := os.Lstat(path); os.IsNotExist(err) {
 			return "", err
 		}
-		if _, err := os.Lstat(filepath.Join(path, solutionFilename)); err == nil {
+		if _, err := os.Lstat(filepath.Join(path, metadataFilepath)); err == nil {
+			return path, nil
+		}
+		if _, err := os.Lstat(filepath.Join(path, legacySolutionFilename)); err == nil {
```

Comment:
Is it ok that above we're using the name `metadataFilepath` but here, `legacySolutionFilename`? Will we presumably clean up all related incoherencies in the future rename PR?

Reply:
> Will we presumably clean up all related solution/metadata incoherencies in the future rename PR?

Yeah, I think it makes sense to do that separately.